api/opencl_runtime_layer.asciidoc

```diff
@@ -235,6 +235,7 @@ Otherwise, it returns one of the following errors:
     by the OpenCL implementation on the device.
   * {CL_OUT_OF_HOST_MEMORY} if there is a failure to allocate resources
     required by the OpenCL implementation on the host.
+  * {CL_INVALID_OPERATION} if _device_ does not support a replacable default On-Device Queue.
```

Comment:
Thanks, this error description looks good, but could you please move it textually up a few lines, so it is listed with the other errors for _device_?

One other consideration: I'm fine listing it separately, but I'd also consider combining this error description with the error description for when _device_ does not support any form of On-Device Queues, if you think that makes sense.

Reply:
Will do, I did consider moving it up textually to be alongside similar errors but wasn't sure if newly added errors had to go at the end.
I'd rather list it separately. Should we make the special case more explicit though, like: "{CL_INVALID_OPERATION} if _device_ does not support a replaceable default On-Device Queue, even when it supports On-Device Queues. "

Reply:
I don't think "even when..." part is necessary.

I actually think we should replace the previous error message vs. adding a new error condition since since the reason to return the error is because the device does not support a replaceable default On-Device Queue and is hence orthogonal to On-Device Queues generally.

It's also redundant, since we already require:

> If CL_DEVICE_QUEUE_REPLACEABLE_DEFAULT is set, CL_DEVICE_QUEUE_SUPPORTED must also be set.